stl/inc/sstream

```diff
@@ -38,6 +42,37 @@ public:
         _Init(_Str.c_str(), _Str.size(), _Getstate(_Mode));
     }
 
+#if _HAS_CXX20
+    explicit basic_stringbuf(const _Alloc& _Al_) : basic_stringbuf(ios_base::in | ios_base::out, _Al_) {}
+
+    basic_stringbuf(ios_base::openmode _Mode, const _Alloc& _Al_)
+        : _Seekhigh(nullptr), _Mystate(_Getstate(_Mode)), _Al(_Al_) {}
+
+    explicit basic_stringbuf(_Mystr&& _Str, ios_base::openmode _Mode = ios_base::in | ios_base::out)
+        : _Al(_STD move(_Str._Getal())) {
+        _Init_string_inplace(_STD move(_Str), _Getstate(_Mode));
+    }
+
+    template <_Allocator _Alloc2>
+    basic_stringbuf(const basic_string<_Elem, _Traits, _Alloc2>& _Str, ios_base::openmode _Mode, const _Alloc& _Al_)
+        : _Al(_Al_) {
+        _Init(_Str.c_str(), _Str.size(), _Getstate(_Mode));
+    }
+
+    template <_Allocator _Alloc2>
+    basic_stringbuf(const basic_string<_Elem, _Traits, _Alloc2>& _Str, const _Alloc& _Al_)
+        : basic_stringbuf(_Str, ios_base::in | ios_base::out, _Al_) {}
+
+    template <_Allocator _Alloc2>
+    explicit basic_stringbuf(
+        const basic_string<_Elem, _Traits, _Alloc2>& _Str, ios_base::openmode _Mode = ios_base::in | ios_base::out)
+        : basic_stringbuf(_Str, _Mode, _Alloc()) {}

```

Comment:
Oh, yes. I used `_Alloc()` only because other parts like
https://github.com/microsoft/STL/blob/2b4cf99c044176637497518294281046439a1bcc/stl/inc/xstring#L2375
using `_Alloc()`.